src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -12688,5 +12942,640 @@ public void RefStructHandler_DynamicInHole(string expression)
+        [InlineData("")]
+        public void ArgumentsOnLocalFunctions_01(string mod)
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+M($"""");
+
+" + mod + @" void M([InterpolatedStringHandlerArgument("""")] CustomHandler c) { }
+";
+
+            var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerArgumentAttribute, GetInterpolatedStringCustomHandlerType("CustomHandler", "struct", useBoolReturns: false) });
+            comp.VerifyDiagnostics(
+                // (4,3): error CS8949: The InterpolatedStringHandlerArgumentAttribute applied to parameter 'CustomHandler' is malformed and cannot be interpreted. Construct an instance of 'CustomHandler' manually.
+                // M($"");
+                Diagnostic(ErrorCode.ERR_InterpolatedStringHandlerArgumentAttributeMalformed, @"$""""").WithArguments("CustomHandler", "CustomHandler").WithLocation(4, 3),
+                // (6,10): error CS8944: 'M(CustomHandler)' is not an instance method, the receiver cannot be an interpolated string handler argument.
+                //  void M([InterpolatedStringHandlerArgument("")] CustomHandler c) { }
+                Diagnostic(ErrorCode.ERR_NotInstanceInvalidInterpolatedStringHandlerArgumentName, @"InterpolatedStringHandlerArgument("""")").WithArguments("M(CustomHandler)").WithLocation(6, 10 + mod.Length)
+            );
+        }
+
+        [Theory]
+        [InlineData("static")]
+        [InlineData("")]
+        public void ArgumentsOnLocalFunctions_02(string mod)
+        {
+            var code = @"
+using System;
+using System.Runtime.CompilerServices;
+
+M(1, $"""");
+
+" + mod + @" void M(int i, [InterpolatedStringHandlerArgument(""i"")] CustomHandler c) => Console.WriteLine(c.ToString());
+
+partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, int i) : this(literalLength, formattedCount) => _builder.Append(""i:"" + i.ToString());
+}
+";
+
+            var verifier = CompileAndVerify(new[] { code, InterpolatedStringHandlerArgumentAttribute, GetInterpolatedStringCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: false) }, expectedOutput: @"i:1");
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("<top-level-statements-entry-point>", @"
+{
+  // Code size       17 (0x11)
+  .maxstack  4
+  .locals init (int V_0)
+  IL_0000:  ldc.i4.1
+  IL_0001:  stloc.0
+  IL_0002:  ldloc.0
+  IL_0003:  ldc.i4.0
+  IL_0004:  ldc.i4.0
+  IL_0005:  ldloc.0
+  IL_0006:  newobj     ""CustomHandler..ctor(int, int, int)""
+  IL_000b:  call       ""void Program.<<Main>$>g__M|0_0(int, CustomHandler)""
+  IL_0010:  ret
+}
+");
+        }
+
+        [Theory]
+        [InlineData("static")]
+        [InlineData("")]
+        public void ArgumentsOnLambdas_01(string mod)
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+var a = " + mod + @" ([InterpolatedStringHandlerArgument("""")] CustomHandler c) => { };
```

Comment:
Are the rules for lambda parameters covered by the spec?

Reply:
There's nothing specific about lambdas in the spec. However, I don't think there needs to be in the interpolated strings proposal: there's nothing particularly different here than for any other attribute that affects the end result, such as `MaybeNull` or similar.

Reply:
1. Regarding the warning: We don't warn for `[MaybeNull]` on lambdas. The interpolated string spec doesn't say anything either.
2. Regarding the error: The spec should probably say that using the argument attribute with "" in a context where there is no receiver is an error. I couldn't find any language to that effect.

Reply:
1. We don't need to spec warnings. `MaybeNull` would also be good to warn on IMO, but since this one affects codgen and `MaybeNull` doesn't, this is a higher priority in my mind.
2. The spec doesn't really say anything about malformed attributes, either how the language interprets assemblies with bad data or how it works in source. We _could_ specify this, but IMO it's not particularly necessary. Invalid data is essentially undefined behavior, and the behavior we chose here is to not allow it.

Reply:
1. I'm not debating whether a warning should be produced here (we intentionally didn't warn for misuse of nullability attributes), just whether we should document it. You're right that we often don't spec warnings. Up to you.
2. I feel stronger about the error though. It's not just a bad metadata behavior, so I think we should spec.
It's not blocking this PR, but seems a simple/quick doc change.